src/modules/edc-demo/components/contract-agreement-transfer-dialog/model/contract-agreement-datasink-form-builder.ts

```diff
@@ -0,0 +1,93 @@
+import {Injectable} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {switchDisabledControls} from '../../../utils/form-group-utils';
+import {jsonValidator} from '../../../validators/json-validator';
+import {urlValidator} from '../../../validators/url-validator';
+import {DataAddressType} from '../../data-address-type-select/data-address-type';
+import {
+  ContractAgreementDatasinkFormModel,
+  ContractAgreementDatasinkFormValue,
+} from './contract-agreement-datasink-form-model';
+import {HttpDatasinkAuthHeaderType} from './http-datasink-auth-header-type';
+import {HttpDatasinkHeaderFormModel} from './http-datasink-header-form-model';
+
+@Injectable()
+export class ContractAgreementDatasinkFormBuilder {
```

Comment:
To unnest the nested form gorup, all this code should be in ContractAgreementTransferDialogForm